pkg/controller/controller_test.go

```diff
@@ -121,6 +122,56 @@ func TestHttpExecutor_TimeBudgetCompletion(t *testing.T) {
 	})
 }
 
+func TestControllerWithTransformationErrorContinueOnError(t *testing.T) {
+	req, err := requests.FromStr("http//localhost:4321/test?test=true", "GET")
+
+	requestExecutor := NewMockRequestExecutor()
+
+	src := source.NewMockSource(req, req, req, req, req, req, req, req)
+
+	transformations := []transformation.Transformation{
+		transformation.NewRemoveQueryFilters(transformation.RemoveQueryParamsTransformConfiguration{
+			Fields: []string{"test"},
+		}),
+		NewMockTransformation(func(request requests.Request) (requests.Request, error) {
+			return req, errors.New("error applying transformation")
+		}),
+	}
+
+	contrl := NewController(src, transformations, requestExecutor, Config{
+		Concurrency:     3,
+		ContinueOnError: true,
+	}, testLogger)
+
+	err = contrl.Execute(context.TODO())
+	require.NoError(t, err)
+
+	require.Len(t, requestExecutor.requests, 0)
+}
+
+func TestControllerWithTransformationError(t *testing.T) {
+	req, err := requests.FromStr("http//localhost:4321/test?test=true", "GET")
+
+	requestExecutor := NewMockRequestExecutor()
+
+	src := source.NewMockSource(req, req, req, req, req, req, req, req)
+
+	transformations := []transformation.Transformation{
+		NewMockTransformation(func(request requests.Request) (requests.Request, error) {
+			return req, errors.New("error applying transformation")
+		}),
+	}
+
+	contrl := NewController(src, transformations, requestExecutor, Config{
+		Concurrency:     3,
+		ContinueOnError: false,
+	}, testLogger)
+
+	err = contrl.Execute(context.TODO())
+	require.Error(t, err)
+	require.Len(t, requestExecutor.requests, 0)
```

Comment:
ma in questo caso che `ContinueOnError: false` non dovrebbe essere più di `0`? sennò mi pare uguale all'altro test

Reply:
No perchè danno tutte errore quindi nessuna chiamata viene effettuata